gpkit/nomials/math.py

```diff
@@ -715,27 +639,26 @@ def __init__(self, left, right):
 
     def as_posyslt1(self, substitutions=None):
         "Returns the posys <= 1 representation of this constraint."
-        # TODO: deal with substitutions
+        # TODO: check if it would be a monomial equality after substitutions
         raise InvalidGPConstraint("SignomialEquality could not simplify"
                                   " to a PosynomialInequality; try calling"
                                   "`.localsolve` instead of `.solve` to"
                                   " form your Model as a SignomialProgram")
 
     def as_gpconstr(self, x0, substitutions=None):
         "Returns GP approximation of an SP constraint at x0"
-        # TODO: deal with substitutions
+        # TODO: check if it would be a monomial equality after substitutions
         siglt0, = self.unsubbed
         posy, negy = siglt0.posy_negy()
         # assume unspecified variables have a value of 1.0
-        x0.update({vk: 1.0 for vk in siglt0.varlocs if vk not in x0})
+        x0.update({vk: 1.0 for vk in siglt0.vks if vk not in x0})
         mec = (posy.mono_lower_bound(x0) == negy.mono_lower_bound(x0))
-        mec.substitutions = self.substitutions
         return mec
 
     def as_approxslt(self):
         "Returns posynomial-less-than sides of a signomial constraint"
-        self._siglt0, = self.unsubbed  # pylint: disable=attribute-defined-outside-init
-        self._posy, self._negy = self._siglt0.posy_negy()  # pylint: disable=attribute-defined-outside-init
+        siglt0, = self.unsubbed
+        self._posy, self._negy = siglt0.posy_negy()  # pylint: disable=attribute-defined-outside-init
```

Comment:
maybe a better question would have been, why is `Monomial(1), Monomial(1)` the right return value?